lib/server.js

```diff
@@ -117,8 +119,29 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
     routes.notify, brRest.when.prefers.ld,
     validate('continuity-server.notification'),
     asyncHandler(async (req, res) => {
-      // FIXME: we need to authenticate `peerId`; http-signatures should be
-      // added here
+      const {url, method, headers} = req;
+      const expectedHeaders = [
+        '(created)',
+        // (key-id) will include the peerId in the signature itself
+        '(key-id)',
+        '(request-target)',
+        'digest',
+        'content-type'
+      ];
+      const parsed = parseRequest(
+        {url, method, headers}, {headers: expectedHeaders});
+      // get parsed parameters from from HTTP header and generate signing string
+      const {keyId, signingString, params: {signature: b64Signature}} = parsed;
+      const key = await wrapKeyPair({peerId: keyId});
+      const verifier = key.verifier();
+      const encoder = new TextEncoder();
+      const data = encoder.encode(signingString);
+      const signature = Buffer.from(b64Signature, 'base64');
+      // verify HTTP signature
+      const verified = await verifier.verify({data, signature});
+      if(!verified) {
+        return res.status(401).end();
```

Comment:
This needs to be 403. We'd only send a 401 if there was no http signature (vs. it not verifying) -- and if we send a 401, we also have to send a `WWW-Authenticate` header. So, let's just send 403.

Reply:
should be addressed here: https://github.com/digitalbazaar/bedrock-ledger-consensus-continuity/pull/249/commits/2341a04e262986a6d37bfc9c52da3264f595a20a